clients/fossid-webapp/src/main/kotlin/LoggingInterceptor.kt

```diff
@@ -0,0 +1,147 @@
+                requestBody.writeTo(buffer)
+
+                val contentType = requestBody.contentType()
+                val charset = contentType?.charset(StandardCharsets.UTF_8) ?: StandardCharsets.UTF_8
+
+                if (request.header("Content-Type") != "application/octet-stream") {
+                    val requestContent = buffer.readString(charset)
+                    append(requestContent.sanitizeForLogging())
+                    append("--> END ${request.method} (${requestBody.contentLength()}-byte body)")
+                } else {
+                    append(
+                        "--> END ${request.method} (binary ${requestBody.contentLength()}-byte body omitted)"
+                    )
+                }
+            }
+        }
+
+        logger.info { requestLogMessage }
+
+        val response = runCatching {
+            chain.proceed(request)
+        }.onFailure {
+            logger.info(it) { "<-- HTTP FAILED" }
+        }.getOrThrow()
+
+        val contentLength = response.body.contentLength()
+        val bodySize = if (contentLength != -1L) "$contentLength-byte" else "unknown-length"
+        val responseMessage = if (response.message.isEmpty()) "" else " ${response.message}"
+
+        val responseLogMessage = buildString {
+            append("<-- ${response.code}$responseMessage ${response.request.url}, $bodySize body\"")
+
+            val source = response.body.source()
+            source.request(Long.MAX_VALUE) // Buffer the entire body.
+            var buffer = source.buffer
+
+            var gzippedLength: Long? = null
+            if ("gzip".equals(request.header("Content-Encoding"), ignoreCase = true)) {
+                gzippedLength = buffer.size
+                GzipSource(buffer.clone()).use { gzippedResponseBody ->
+                    buffer = Buffer()
+                    buffer.writeAll(gzippedResponseBody)
+                }
+            }
+
+            val contentType = response.body.contentType()
+            val charset = contentType?.charset(StandardCharsets.UTF_8) ?: StandardCharsets.UTF_8
+
+            if (!response.promisesBody()) {
+                append("<-- END HTTP")
+            } else if (response.header("Content-Type") == "application/octet-stream") {
+                append("<-- END HTTP (binary ${buffer.size}-byte body omitted)")
+            } else {
+                append(buffer.clone().readString(charset))
+
+                if (gzippedLength != null) {
+                    append("<-- END HTTP (${buffer.size}-byte, $gzippedLength-gzipped-byte body)")
+                } else {
+                    append("<-- END HTTP (${buffer.size}-byte body)")
+                }
+            }
+        }
+
+        logger.info { responseLogMessage }
+
+        return response
+    }
+
+    /**
+     * Sanitize the string for logging purposes by removing sensitive information like credentials.
+     */
+    private fun String.sanitizeForLogging(): String {
```

Comment:
Sorry, yet more stuff as I find the time to actually checkout the code: This function does not require class context and should be moved to the top level below this class.